libs/libopenfpgacapnproto/CMakeLists.txt

```diff
@@ -0,0 +1,99 @@
+include(GNUInstallDirs)
+
+if(NOT MSCV)
+    # These flags generate noisy but non-bug warnings when using lib kj,
+    # supress them.
+    set(WARN_FLAGS_TO_DISABLE
+        -Wno-undef
+        -Wno-non-virtual-dtor
+        )
+    foreach(flag ${WARN_FLAGS_TO_DISABLE})
+        CHECK_CXX_COMPILER_FLAG(${flag} CXX_COMPILER_SUPPORTS_${flag})
+        if(CXX_COMPILER_SUPPORTS_${flag})
+            #Flag supported, so enable it
+            add_compile_options(${flag})
+        endif()
+    endforeach()
+endif()
+
+# Create generated headers from capnp schema files
+#
+# Each schema used should appear here.
+
+set(CAPNP_DEFS
+    gen/unique_blocks_uxsdcxx.capnp
+)
+
+capnp_generate_cpp(CAPNP_SRCS CAPNP_HDRS
+    ${CAPNP_DEFS}
+)
+
+
+
+find_program(WGET wget REQUIRED)
+find_package(ZLIB REQUIRED)
+
+# Add Java schema
+set(JAVA_SCHEMA ${CMAKE_CURRENT_BINARY_DIR}/schema/capnp/java.capnp)
+add_custom_command(
+    OUTPUT ${JAVA_SCHEMA}
+    COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_CURRENT_BINARY_DIR}/schema/capnp/
+    COMMAND ${WGET}
+        https://raw.githubusercontent.com/capnproto/capnproto-java/master/compiler/src/main/schema/capnp/java.capnp
+        -O ${JAVA_SCHEMA}
+)
+
+add_custom_target(
+    get_java_capnp_schema_openfpga
+    DEPENDS ${JAVA_SCHEMA}
+)
+
+set(CAPNPC_IMPORT_DIRS)
+list(APPEND CAPNPC_IMPORT_DIRS ${CMAKE_CURRENT_BINARY_DIR}/schema)
+
+
+install(FILES ${CAPNP_DEFS} DESTINATION ${CMAKE_INSTALL_DATADIR}/openfpga)
+
+add_library(libopenfpgacapnproto STATIC
+            ${CAPNP_SRCS}
+            ${IC_SRCS}
+            mmap_file.h
+            mmap_file.cpp
+            serdes_utils.h
+            serdes_utils.cpp
+            )
+
+
+add_dependencies(libopenfpgacapnproto
+    get_java_capnp_schema_openfpga
+    generate_unique_block_capnp
+    )
+
+
+target_include_directories(libopenfpgacapnproto PUBLIC
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${CMAKE_CURRENT_BINARY_DIR}
+    ${CMAKE_CURRENT_BINARY_DIR}/gen
+    )
+target_link_libraries(libopenfpgacapnproto
+    libopenfpgautil
```

Comment:
Add the capnproto library from vpr: libvtrcapnproto as one of the target link library. You can avoid to duplicate the files mmap_file.cpp|h and serdes_utils.cpp|h